src/lsp.ts

```diff
@@ -29,6 +29,9 @@ function createClient(jarPath: string): LanguageClient {
         },
         middleware: {
             handleDiagnostics(uri, diagnostics, next) {
+                if (!state.config().displayDiagnostics) {
+                    return;
```

Comment:
With this, I noticed when the setting is unchecked, whatever linter warnings existed before are not cleared, even when the user types in the relevant files. If we `return next(uri, [])` then the lint warnings will be cleared when the file is changed (and maybe other events like reopening it).

I'll make this change if you don't have time, just didn't want to modify your PR without checking with you.

Reply:
Feel free to run with it!

Reply:
And thanks a lot!